No sound on macOS

OS version: 14.1.1 (Sonoma)

Running the to buffer or not to buffer example does not play any sound when  enter is struck. No errors are displayed, it's simply silent.

Found a workaround!

TL;DR - manually set purego version to v0.7.0 after `go get github.com/gopxl/beep/speaker`

1. First, I cleared my `go.mod` dependencies file.

2. run code below
```
package main

import (
	"fmt"

	_ "github.com/gopxl/beep/speaker"
)

func init() {
	fmt.Println("init()")
}

func main() {
	fmt.Println("main()")
}
```

3. The following happenend in the terminal:
```
$  go run -v -a .
main.go:6:2: no required module provides package github.com/gopxl/beep/speaker; to add it:
        go get github.com/gopxl/beep/speaker
$ go get github.com/gopxl/beep/speaker
go: added github.com/ebitengine/oto/v3 v3.1.0
go: added github.com/ebitengine/purego v0.5.0
go: added github.com/gopxl/beep v1.4.0
go: added github.com/pkg/errors v0.9.1
go: added golang.org/x/sys v0.12.0
dominggus@dombp15 .../projects/doWork/go-playground  go run -v -a .                      
internal/unsafeheader
internal/goos
internal/godebugs
internal/coverage/rtcov
internal/goexperiment
internal/goarch
runtime/internal/atomic
internal/cpu
internal/itoa
internal/chacha8rand
runtime/internal/math
internal/abi
runtime/internal/sys
math/bits
unicode/utf8
internal/race
sync/atomic
cmp
unicode
github.com/ebitengine/purego/internal/strings
internal/bytealg
slices
math
runtime
internal/reflectlite
sync
internal/testlog
runtime/cgo
errors
sort
path
internal/oserror
io
internal/safefilepath
strconv
syscall
bytes
strings
reflect
regexp/syntax
internal/syscall/execenv
internal/syscall/unix
time
regexp
io/fs
internal/poll
internal/fmtsort
os
github.com/ebitengine/purego
fmt
github.com/ebitengine/oto/v3/internal/mux
github.com/ebitengine/purego/objc
github.com/pkg/errors
github.com/gopxl/beep
github.com/ebitengine/oto/v3
github.com/gopxl/beep/speaker
example.com/mod
^Csignal: interrupt
```

4. Then it hangs as expected. Then I open the `go.mod` file again:
```
module example.com/mod

go 1.22.1

require (
	github.com/ebitengine/oto/v3 v3.1.0 // indirect
	github.com/ebitengine/purego v0.5.0 // indirect
	github.com/gopxl/beep v1.4.0 // indirect
	github.com/pkg/errors v0.9.1 // indirect
	golang.org/x/sys v0.12.0 // indirect
)
```
5. Then, I change purego's version to `v0.7.0`

6. Voila, it works!
```
go run -v -a .
internal/goarch
internal/goexperiment
internal/coverage/rtcov
internal/unsafeheader
internal/godebugs
internal/goos
internal/cpu
internal/abi
runtime/internal/atomic
runtime/internal/math
internal/itoa
internal/chacha8rand
runtime/internal/sys
math/bits
unicode/utf8
internal/race
sync/atomic
unicode
cmp
github.com/ebitengine/purego/internal/cgo
github.com/ebitengine/purego/internal/strings
internal/bytealg
slices
math
runtime
internal/reflectlite
sync
internal/testlog
runtime/cgo
errors
sort
internal/oserror
internal/safefilepath
path
io
strconv
syscall
bytes
strings
reflect
regexp/syntax
internal/syscall/execenv
time
internal/syscall/unix
regexp
io/fs
internal/poll
internal/fmtsort
os
fmt
github.com/ebitengine/oto/v3/internal/mux
github.com/pkg/errors
github.com/gopxl/beep
github.com/ebitengine/purego
github.com/ebitengine/purego/objc
github.com/ebitengine/oto/v3
github.com/gopxl/beep/speaker
example.com/mod
init()
main()
```

--- a/go.mod
+++ b/go.mod
@@ -15,7 +15,7 @@ require (
 
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.5.0 // indirect
+	github.com/ebitengine/purego v0.7.1 // indirect
 	github.com/gdamore/encoding v1.0.0 // indirect
 	github.com/icza/bitio v1.1.0 // indirect
 	github.com/jfreymuth/vorbis v1.0.2 // indirect

